README.md

```diff
@@ -123,6 +123,31 @@ baseImageOverrides:
   github.com/my-user/my-repo/cmd/foo: registry.example.com/base/for/foo
 ```
 
+### Overriding Go build settings
+
+By default, `ko` builds the binary with no additional arguments other than `--trimpath` (if available) and the ones required to name the binary accordingly. You can extend the `go build` arguments by providing extra build flags using a [GoReleaser](https://github.com/goreleaser/goreleaser) influenced `builds` configuration section in your `.ko.yaml`.
```

Comment:
I haven't fully thought through this idea, but:

It would be nice if the default build config (I'm thinking of `--trimpath` specifically here) were expressed in terms of this goreleaser config thing, and supplying your own build args (via the `builds` clause in `.ko.yaml`) would override it completely. This would give you a way to actually drop the `--trimpath` flag if you don't want it.

We would basically have good defaults for builds but offer the flexibility to do whatever you want. We can document the default or (better) have some way to print the default `.ko.yaml` file so folks can fiddle with it.

Reply:
It is a good point. I guess the tricky part is how to document it properly so that users understand to put `--trimpath` in there and not expect it to be automatically inserted as before. Will change that.